### DISCUSSION TITLE
one-click-installers-tts-6.0 audiocraft, fastapi-cli, fairseq, vector-quantize version conflict

### DISCUSSION THREAD

**Discussion by cyberbrain666666** [rsxdalv/TTS-WebUI#312] [Q&A] [answered]
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
audiocraft 1.3.0a0 requires torch==2.1.0, but you have torch 2.0.0 which is incompatible.
fastapi-cli 0.0.3 requires typer>=0.12.3, but you have typer 0.9.4 which is incompatible.
audiocraft 1.3.0a0 requires hydra-core>=1.1, but you have hydra-core 1.0.7 which is incompatible.
audiocraft 1.3.0a0 requires torch==2.1.0, but you have torch 2.0.0 which is incompatible.
fairseq 0.12.4 requires hydra-core<1.1,>=1.0.7, but you have hydra-core 1.3.2 which is incompatible.
fairseq 0.12.4 requires omegaconf<2.1, but you have omegaconf 2.3.0 which is incompatible.
vector-quantize-pytorch 1.14.22 requires einops>=0.8.0, but you have einops 0.7.0 which is incompatible.

**Comment by rsxdalv** [comment, marked as answer]
Yes because pip does not want these projects to work together, but in reality they do.
It's also because these projects haven't listed all of the versions that actually work.
And audiocraft just wants to upgrade to newer torch despite working perfectly fine with 2.0.0

**Comment by cyberbrain666666** [comment]
So everything works fine ?

**Reply by rsxdalv** [reply]
Yeah, if something doesn't work feel free to report. Maybe someday in the far future I will have enough resources to force all of these projects to be reasonable. But I think this "error" will be here for years to come.

**Comment by cyberbrain666666** [comment]
After installing it was running fine but after turning it off and every time turning it back on, a message pops up 
requests.exceptions.ConnectionError:
HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /startup-events (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000002353C0059F0>: Failed to establish new connection: [WinError 10061] Unable to establish connection because target computer is actively rejecting it'))
It only goes by a heavily simplified version http://localhost:3000

TTS rejects http://localhost:7860/
It only goes through a heavily simplified version 
http://localhost:3000

**Comment by rsxdalv** [comment]
That's new. Did you change the default port to be 80? Does it not work or
just report this error while still being functional?
I'm guessing that it didn't close the connection but that is also abnormal.
I am able to close and reopen as many times as I like.

**Reply by cyberbrain666666** [reply]
I didn't change anything I can't do anything the server doesn't work.

Press any key to continue . . .
> tts-generation-webui-react@0.1.0 start
> next start

- ready started server on 0.0.0.0:3000, url: http://localhost:3000
gradio api handler rvc_model_reload
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11457:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async resolve_config (D:\TTS-Generation-Webui\one-click-installers-tts-6.0\tts-generation-webui\react-ui\.next\server\chunks\327.js:1029:20)
    at async D:\TTS-Generation-Webui\one-click-installers-tts-6.0\tts-generation-webui\react-ui\.next\server\chunks\327.js:353:18 {
  cause: Error: connect ECONNREFUSED 127.0.0.1:7860
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
    errno: -4078,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 7860
  }
}
gradio api handler rvc_index_reload
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11457:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async resolve_config (D:\TTS-Generation-Webui\one-click-installers-tts-6.0\tts-generation-webui\react-ui\.next\server\chunks\327.js:1029:20)
    at async D:\TTS-Generation-Webui\one-click-installers-tts-6.0\tts-generation-webui\react-ui\.next\server\chunks\327.js:353:18 {
  cause: Error: connect ECONNREFUSED 127.0.0.1:7860
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
    errno: -4078,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 7860
  }
}
gradio api handler rvc_model_reload
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11457:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async resolve_config (D:\TTS-Generation-Webui\one-click-installers-tts-6.0\tts-generation-webui\react-ui\.next\server\chunks\327.js:1029:20)
    at async D:\TTS-Generation-Webui\one-click-installers-tts-6.0\tts-generation-webui\react-ui\.next\server\chunks\327.js:353:18 {
  cause: Error: connect ECONNREFUSED 127.0.0.1:7860
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
    errno: -4078,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 7860
  }
}
gradio api handler rvc_index_reload
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11457:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async resolve_config (D:\TTS-Generation-Webui\one-click-installers-tts-6.0\tts-generation-webui\react-ui\.next\server\chunks\327.js:1029:20)
    at async D:\TTS-Generation-Webui\one-click-installers-tts-6.0\tts-generation-webui\react-ui\.next\server\chunks\327.js:353:18 {
  cause: Error: connect ECONNREFUSED 127.0.0.1:7860
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
    errno: -4078,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 7860

**Comment by rsxdalv** [comment]
Got it, thanks. Are you running another webui at localhost:7860?

**Comment by cyberbrain666666** [comment]
Not

**Comment by rsxdalv** [comment]
It seems that the main UI is failing to start for some reason. Do you have
the full log from when you start it?

**Reply by cyberbrain666666** [reply]
Loading extensions:
  Loaded extension: callback_save_generation_ffmpeg
  Loaded extension: callback_save_generation_ffmpeg_flac
  Loaded extension: callback_save_generation_musicgen_ffmpeg
  Loaded extension: callback_save_generation_musicgen_ffmpeg_flac
  Loaded extension: empty_extension
  Loaded 3 callback_save_generation extensions.
  Loaded 2 callback_save_generation_musicgen extensions.
D:\TTS-Generation-Webui\one-click-installers-tts-6.0\installer_files\env\lib\site-packages\transformers\deepspeed.py:23: FutureWarning: transformers.deepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
D:\TTS-Generation-Webui\one-click-installers-tts-6.0\installer_files\env\lib\site-packages\gradio\components\dropdown.py:231: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: None or set allow_custom_value=True.
  warnings.warn(
[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\cyber\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
Starting Gradio server...
 Gradio interface options:
  inline:               False
  inbrowser:            True
  share:                False
  debug:                False
  max_threads:          41
  auth:                 None
  auth_message:         None
  prevent_thread_lock:  False
  show_error:           False
  server_name:          0.0.0.0
  server_port:          0
  show_tips:            False
  height:               500
  width:                100%
  favicon_path:         None
  ssl_keyfile:          None
  ssl_certfile:         None
  ssl_keyfile_password: None
  ssl_verify:           True
  quiet:                True
  show_api:             True
  file_directories:     None
  _frontend:            True
Running on local URL:  http://0.0.0.0:0

------------------------------------------------------------------------------------------------------------------------
adapters.py 486 send
resp = conn.urlopen(

connectionpool.py 844 urlopen
retries = retries.increment(

retry.py 515 increment
raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]

urllib3.exceptions.MaxRetryError:
HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /startup-events (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000001FE925A1B10>: Failed to establish a new connection: [WinError 10061] The connection cannot be established because the target computer actively refuses it'))
------------------------------------------------------------------------------------------------------------------------
server.py 285 <module>
start_server()

server.py 276 start_server
demo.queue(

blocks.py 2071 launch
requests.get(f"{self.local_url}startup-events", verify=ssl_verify)

api.py 73 get
return request("get", url, params=params, **kwargs)

api.py 59 request
return session.request(method=method, url=url, **kwargs)

sessions.py 589 request
resp = self.send(prep, **send_kwargs)

sessions.py 703 send
r = adapter.send(request, **kwargs)

adapters.py 519 send
raise ConnectionError(e, request=request)

requests.exceptions.ConnectionError:
HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /startup-events (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000001FE925A1B10>: Failed to establish a new connection: [WinError 10061] The connection cannot be established because the target computer actively refuses it'))

Done!
Press any key to continue . . .
> tts-generation-webui-react@0.1.0 start
> next start

- ready started server on 0.0.0.0:3000, url: http://localhost:3000

**Reply by cyberbrain666666** [reply]
I don't know if it's affected by this error, but I pasted into the bat file for a while the command set PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.6,max_split_size_mb:128 
Which I found this on the web as a solution to the memory overflow ( RuntimeError: CUDA out of memory)connection as I generated longer audio files. However, I removed this command rally so it is unlikely to be affected.

**Comment by rsxdalv** [comment]
In config.json please set the port to 7860 or if you don't have any special
settings you can just delete the config.json file and it will reset to
default.

**Comment by cyberbrain666666** [comment]
I removed the file config.json from the virtual env D:\TTS-Generation-Webui\one-click-installers-tts-6.0\installer_files\env\Lib\site-packages\rvc_pkg\configs\config.json
And still the same.
The file has not been re-created

Load
Loading extensions:
  Loaded extension: callback_save_generation_ffmpeg
  Loaded extension: callback_save_generation_ffmpeg_flac
  Loaded extension: callback_save_generation_musicgen_ffmpeg
  Loaded extension: callback_save_generation_musicgen_ffmpeg_flac
  Loaded extension: empty_extension
  Loaded 3 callback_save_generation extensions.
  Loaded 2 callback_save_generation_musicgen extensions.
 Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
D:TTS-Generation-Webui\one-click-installers-tts-6.0\installer_files\env\lib\site-packages\gradio\components\dropdown.py:231: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: None or set allow_custom_value=True.
  warnings.warn(
[nltk_data] Downloading package point to
[nltk_data] C:³³³³AppData³³³...
[nltk_data] Package point is already up-to-date!
Starting Gradio server...
 Gradio interface options:
  inline:               False
  inbrowser: True
  share:                False
  debug: False
  max_threads:          41
  auth: None
  auth_message: None
  prevent_thread_lock: False
  show_error:           False
  server_name: 0.0.0.0
  server_port: 0
  show_tips: False
  height: 500
  width: 100%
  favicon_path: None
  ssl_keyfile: None
  ssl_certfile: None
  ssl_keyfile_password: None
  ssl_verify: True
  quiet: True
  show_api: True
  file_directories: None
  _frontend: True
Running on local URL:  http://0.0.0.0:0

------------------------------------------------------------------------------------------------------------------------
adapters.py 486 send
resp = conn.urlopen(

connectionpool.py 844 urlopen
retries = retries.increment(

retries.py 515 increment
raise MaxRetryError(_pool, url, reason) from reason # type: ignore[arg-type]

urllib3.exceptions.MaxRetryError:
HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /startup-events (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000002B758CF1600>: Failed to establish a new connection: [WinError 10061] Unable to establish connection because target computer is actively refusing it'))
------------------------------------------------------------------------------------------------------------------------
server.py 285 <module>
start_server()

server.py 276 start_server
demo.queue(

blocks.py 2071 launch
requests.get(f“{self.local_url}startup-events”, verify=ssl_verify)

api.py 73 get
return request(“get”, url, params=params, **kwargs)

api.py 59 request
return session.request(method=method, url=url, **kwargs)

sessions.py 589 request
resp = self.send(prep, **send_kwargs)

sessions.py 703 send
r = adapter.send(request, **kwargs)

adapters.py 519 send
raise ConnectionError(e, request=request)

requests.exceptions.ConnectionError:
HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /startup-events (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000002B758CF1600>: Failed to establish a new connection: [WinError 10061] Unable to establish connection because target computer is actively refusing it'))

Done!
Press any key to continue . . .
> tts-generation-webui-react@0.1.0 start
> next start

- ready started server on 0.0.0.0:3000, url: http://localhost:3000
-

**Reply by rsxdalv** [reply]
Sorry for the confusion, the config.json file in the main directory: so for you that would be 
```
D:\TTS-Generation-Webui\one-click-installers-tts-6.0\tts-generation-webui\config.json
```

**Reply by cyberbrain666666** [reply]
Thanks a lot it worked by the way how to solve RuntimeError: CUDA out of memory when generating longer audio files

**Reply by rsxdalv** [reply]
> Thanks a lot it worked by the way how to solve RuntimeError: CUDA out of memory when generating longer audio files

Happy that it worked. As for the memory - which model are you using? The advice depends on that. Also how much RAM/VRAM do you have?

**Reply by cyberbrain666666** [reply]
I have a nvidia rtx 4090 graphics card with 24 GB of vram. I'm talking about music gen + audio gen. Model facebook/musicgen-stereo-melody-large and setting the duration to 120

**Reply by rsxdalv** [reply]
That model is huge, although I think there should be a way for it to generate 120 without OOM. Also I recommend not using MBD unless you know you want it for sure.
Are you able to generate 60 seconds?

**Reply by cyberbrain666666** [reply]
Yes but it's not enough for me. Supposedly, through some kind of setting, this can be circumvented ?

**Reply by rsxdalv** [reply]
Sorry for not replying faster. Can it be? I have finally set up my 24gb 3090 so I could test a similar test case, but if you have any additional information that I could research it would be helpful.
